C-Plus-Plus/sort/Topological_Sort.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+  Topological sorting for Directed Acyclic Graph (DAG) is a linear ordering
+  of vertices such that for every directed edge uv, vertex u comes before v in
+  the ordering. Topological Sorting for a graph is not possible if the graph is
+  not a DAG.
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+vector<vector<int>> adjList;
+stack<int> s;
+bool *visited;
+
+// Number of Test Cases (Vertices)
+int n;
+
+// function to Create and represent a graph by storing data.
+void createGraph()
+{
+    int x, y;
+    adjList.resize(n);
+
+    for (int j = 0; j < n; j++)
+    {
+        cin >> x >> y;
+        adjList[x].push_back(y);
+    }
+}
+
+// A recursive function used by topoSortFun.
+void topoSortFun(int i)
+{
+    visited[i] = true;
+    for (int j = 0; j < adjList[i].size(); j++)
+    {
+        int key = adjList[i][j];
+        if (!visited[key])
+            topoSortFun(key);
+    }
+    s.push(i);
+}
+
+// The function to do Topological Sort. It uses recursive topoSortFun(). 
+void topoSort()
+{
+    memset(visited, false, sizeof(visited));
+
+    for (int i = 0; i < n; i++)
+    {
+        if (!visited[i])
+            topoSortFun(i);
+    }
+}
+void display()
+{
+    for (int i = 0; i < n; i++)
+    {
+        cout << s.top() << " ";
+        s.pop();
+    }
+}
+int main()
+{
+    cout<<"Enter number of node you want to add: ";
+    cin >> n;
+    visited = new bool[n];
+    cout<<"Enter nodes of graph: ";
+    createGraph();
+    topoSort();
+    cout<<"Graph after topological sort: ";
+    display();
+}
+
+/*
+Test Case:
+Input:
```

Comment:
Update your sample I/O according to your code. The sample I/O should be the same as it will look on the terminal when someone runs the program.